src/external/visualization_integration.cu

```diff
@@ -1,35 +1,84 @@
 #include <iostream>
 #include <fstream>
 #include <string>
+#include <vector>
 
 #include "visualization_integration.cuh"
 #include "../simulation_objects/Particle.cuh"
 
 
-__host__ std::string get_visualization_endpoint()
+__host__ std::pair<std::string, std::string> get_visualization_endpoints()
 {
     std::ifstream f("config/visualization_endpoint.txt", std::ios::in);
-    std::string url;
-    getline(f, url);
-    return url;
+    std::string particles_url, poly_url;
+    getline(f, particles_url);
+    getline(f, poly_url);
+    return {particles_url, poly_url};
 }
 
-__host__ std::string vector_double_to_json_array(const std::vector<double> &v)
+template<class T>
+__host__ std::string to_string_extended(const T &v)
 {
-    if(v.empty())
-        return "[]";
+    return std::to_string(v);
+}
+
+template<class T>
+__host__ std::string to_string_extended(const std::vector<T> &v)
+{
+    std::string body = "[";
+    for(int i = 0; i < v.size(); ++i)
+    {
+        body += to_string_extended(v[i]) + ',';
+    }
+    body.back() = ']';
+
+    return body;
+}
+
+__host__ bool send_poly_to_visualization(const std::pair<std::string, std::string> &urls, const Polyhedron *polyhedron)
+{
+    std::vector<double> polyhedron_vertices, polyhedron_faces;
+    std::vector<std::vector<double>> poly;
+
+    int n_faces = polyhedron->get_n_of_faces();
+    int n_vertices_on_face[n_faces];
+    Face* poly_faces = polyhedron->get_faces();
+
+    for (int i = 0; i < n_faces; ++i) {
+        n_vertices_on_face[i] = poly_faces[i].get_n_of_vertices();
+    }
```

Comment:
1. Curly brace should be on a separate line
2. To be honest, I don't like the solution with an array. It requires additional memory, and it may also slightly reduce code readability. We don't need the sizes to be precalculated